gpytorch/kernels/rff_kernel.py

```diff
@@ -128,7 +128,11 @@ def forward(self, x1: Tensor, x2: Tensor, diag: bool = False, last_dim_is_batch:
         if diag:
             return (z1 * z2).sum(-1) / D
         if x1_eq_x2:
-            return LowRankRootLazyTensor(z1 / math.sqrt(D))
+            # Exploit low rank structure, if there are fewer features than data points
+            if z1.size(-1) < z2.size(-2):
```

Comment:
Okay, in my defense for the original version of this who's out there using RFFs with <1000 data points

Reply:
Haha yeah I figure that I should add this, since I was going through the effort to make this code more robust anyways.